src/Paramore.Brighter.Outbox.DynamoDB/DynamoDbOutbox.cs

```diff
@@ -46,6 +51,18 @@ public class DynamoDbOutbox :
         private readonly DynamoDBOperationConfig _dynamoOverwriteTableConfig;
         private readonly Random _random = new Random();
 
+        // Stores context of the current progress of any paged queries for outstanding or dispatched messages to particular topics
```

Comment:
Yeah I'm forever trying to decide whether something would or wouldn't be obvious to another developer coming into the codebase. In these instances I thought it would be helpful as Dynamo's paging mechanism (and hence the need for these variables) can be confusing, and it may not be clear why there's a `ConcurrentDictionary` being used to store what is only ever used as a list. I'm happy to remove the comments if you think that isn't warranted though?

Reply:
I'd comment more on the why in the code, over the state of the object.